src/trace.ml

```diff
@@ -18,6 +18,16 @@ let supports_fzf =
         | Ok () -> true))
 ;;
 
+let create_elf ~executable ~(when_to_snapshot : When_to_snapshot.t) =
+  let elf = Elf.create executable in
+  match when_to_snapshot, elf with
+  | Application_calls_a_function _, None ->
+    Deferred.Or_error.errorf
+      "As far as magic-trace can tell, that executable doesn't have a symbol table. Was \
```

Comment:
I don't think this should be a fatal error condition. For instance, you can imagine not having symbols for the `java` binary, but still get mileage out of magic-trace from the perf map files. Likewise, we may not have symbols for the main binary, but have symbols for dynamically loaded libraries, and still find magic-trace useful for introspection.

Reply:
Oh, hold on, I think I misread this code...